openapi/data_repository_service.swagger.yaml

```diff
@@ -194,57 +181,77 @@ definitions:
           sha512
   Bundle:
     type: object
-    required: ['id', 'object_ids', 'created', 'updated', 'version', 'checksums']
     properties:
       id:
         type: string
-        description: |-
+        description: >-
           An identifier, unique to this Data Bundle
-      object_ids:
-        type: array
-        items:
-          type: string
+      name:
+        type: string
         description: |-
-          The list of Data Objects that this Data Bundle contains.
+          A string that can be optionally used to name a Data Bundle.
+      size:
+        type: string
+        format: int64
+        description: |-
+          The cumulative size of all Data Objects and Bundles listed within
+          contents field in bytes.
       created:
         type: string
         format: date-time
-        description: |-
-          Timestamp of object creation in RFC3339.
+        description: >-
+          Timestamp of Bundle creation in RFC3339.
       updated:
         type: string
         format: date-time
-        description: |-
-          Timestamp of update in RFC3339, identical to create timestamp in systems
-          that do not support updates.
+        description: >-
+          Timestamp of Bundle update in RFC3339, identical to create timestamp in
+          systems that do not support updates.
       version:
         type: string
-        description: |-
-          A string representing a version, some systems may use checksum, a RFC3339
-          timestamp, or incrementing version number. For systems that do not support
-          versioning please use your update timestamp as your version.
+        description: >-
+          A string representing a version, some systems may use checksum, a
+          RFC3339 timestamp, or incrementing version number.
       checksums:
         type: array
+        description: >-
+          At least one checksum must be provided.
+          The Data Bundle checksum is computed over a sorted concatenation of all 
+          the checksums within the top-level 'contents' of the Bundle (not recursive).
```

Comment:
@susheel -- confirming that this checksum does **not** include the `name` field from the `contents` array? (I"m not sure which makes more sense -- as written it means that two bundles that contain the same objects, but with different names, will end up having the same checksum as each other.)

Reply:
Clarified description to confirm names are NOT included in the checksum computation. Yes, two bundles containing the same objects should resolve to have the same checksum. The names assigned by the user should not affect the checksum, just how the individual objects are materialised onto disk.